code/modules/vehicles/mecha/working/ripley.dm

```diff
@@ -99,6 +97,110 @@
 	fire = 100
 	acid = 100
 
+/obj/vehicle/sealed/mecha/ripley/paddy
+	desc = "Autonomous Power Loader Unit Subtype Paddy. A Modified MK-I Ripley design intended for light security use."
+	name = "\improper APLU \"Paddy\""
+	icon_state = "paddy"
+	base_icon_state = "paddy"
+	max_temperature = 20000
+	max_integrity = 250
+	mech_type = EXOSUIT_MODULE_PADDY
+	possible_int_damage = MECHA_INT_FIRE|MECHA_INT_CONTROL_LOST|MECHA_INT_SHORT_CIRCUIT
+	accesses = list(ACCESS_MECH_SCIENCE, ACCESS_MECH_SECURITY)
+	armor_type = /datum/armor/mecha_paddy
+	wreckage = /obj/structure/mecha_wreckage/ripley/paddy
+	silicon_icon_state = "paddy-empty"
+	equip_by_category = list(
+		MECHA_L_ARM = null,
+		MECHA_R_ARM = null,
+		MECHA_UTILITY = list(/obj/item/mecha_parts/mecha_equipment/ejector/seccage),
+		MECHA_POWER = list(),
+		MECHA_ARMOR = list(),
+	)
+	///Siren Lights/Sound State
+	var/siren = FALSE
+	///Overlay for Siren Lights
+	var/mutable_appearance/sirenlights
+	///Looping sound datum for the Siren audio
+	var/datum/looping_sound/siren/weewooloop
+
+/datum/armor/mecha_paddy
+	melee = 40
+	bullet = 20
+	laser = 10
+	energy = 20
+	bomb = 40
+	fire = 100
+	acid = 100
+
+/obj/vehicle/sealed/mecha/ripley/paddy/Initialize(mapload)
+	. = ..()
+	weewooloop = new(src, FALSE, FALSE)
+	weewooloop.volume = 100
+
+/obj/vehicle/sealed/mecha/ripley/paddy/generate_actions()
+	. = ..()
+	initialize_passenger_action_type(/datum/action/vehicle/sealed/mecha/siren)
+
+/obj/vehicle/sealed/mecha/ripley/paddy/mob_exit(mob/M, silent = FALSE, randomstep = FALSE, forced = FALSE)
+	var/obj/item/mecha_parts/mecha_equipment/ejector/seccage/cargo_holder = locate(/obj/item/mecha_parts/mecha_equipment/ejector/seccage) in equip_by_category[MECHA_UTILITY]
+	for(var/mob/contained in cargo_holder)
+		cargo_holder.cheese_it(contained)
+	togglesiren(force_off = TRUE)
+	return ..()
+
+/obj/vehicle/sealed/mecha/ripley/paddy/proc/togglesiren(force_off = FALSE)
+	if(force_off || siren)
+		weewooloop.stop()
+		siren = FALSE
+	else
+		weewooloop.start()
+		siren = TRUE
+	for(var/mob/occupant as anything in occupants)
+		balloon_alert(occupant, "siren [siren ? "activated" : "disabled"]")
+		var/datum/action/act = locate(/datum/action/vehicle/sealed/mecha/siren) in occupant.actions
+		act.button_icon_state = "mech_siren_[siren ? "on" : "off"]"
+		act.build_all_button_icons()
+	update_overlays()
+
+/obj/vehicle/sealed/mecha/ripley/paddy/update_overlays()
+	cut_overlays()
+	. = ..()
+	if(siren)
+		sirenlights = new()
+		sirenlights.icon = icon
+		sirenlights.icon_state = "paddy_sirens"
+		SET_PLANE_EXPLICIT(sirenlights, ABOVE_LIGHTING_PLANE, src)
+		add_overlay(sirenlights)
```

Comment:
```suggestion
	. = ..()
	if(!siren)
		return
	sirenlights = new()
	sirenlights.icon = icon
	sirenlights.icon_state = "paddy_sirens"
	SET_PLANE_EXPLICIT(sirenlights, ABOVE_LIGHTING_PLANE, src)
	. += sirenlights
```

But also why are we holding `sirenlights` in memory in the first place?

Reply:
Copypasted from borg lights, honestly. Also, if we don't call cut_overlays(), the siren lights never go away. It doesn't seem to be done for us up the chain.

Reply:
That's because you are using `add_overlays` instead of just pushing it to the list generated in `update_overlays`
If you add it via `. += whatever` then you don't need to call `cut_overlays`